kinisi/jump_diffusion_analyzer.py

```diff
@@ -31,218 +34,69 @@ class JumpDiffusionAnalyzer(Analyzer):
-
-    @property
-    def mstd(self) -> np.ndarray:
-        """
-        :return: MSTD for the input trajectories. Note that this is the bootstrap sampled MSD, not the numerical
-            average from the data.
+            then it is necessary to identify if these constitute a series of :py:attr:`consecutive` trajectories or
+            a series of :py:attr:`identical` starting points with different random seeds, in which case the `dtype`
+            should be either :py:attr:`consecutive` or :py:attr:`identical`.:
+        :param dt: Time intervals to calculate the displacements over. Optional, defaults to a :py:mod:`scipp` array
+            ranging from the smallest interval (i.e., time_step * step_skip) to the full simulation length, with 
+            a step size the same as the smallest interval.
+        :param dimension: Dimension/s to find the displacement along, this should be some subset of `'xyz'` indicating
+            the axes of interest. Optional, defaults to `'xyz'`.
+        :param distance_unit: The unit of distance in the simulation input. This should be a :py:mod:`scipp` unit and
+            defaults to :py:attr:`sc.units.angstrom`.
+        :param progress: Print progress bars to screen. Optional, defaults to :py:attr:`True`.
+        
+        :returns: The :py:class:`DiffusionAnalyzer` object with the mean-squared displacement calculated.
         """
-        return self._diff.n
-
+        p = super()._from_Xdatcar(trajectory, specie, time_step, step_skip, dtype, dt, dimension, distance_unit, progress)
+        p.mstd = calculate_mstd(p.trajectory, progress)
+        return p
+    
     @property
-    def mstd_std(self) -> np.ndarray:
-        """
-        :return: MSD standard deviation values for the input trajectories (a single standard deviation).
-        """
-        return self._diff.s
+    def n_atoms(self):
+        """Property to access the n_atoms."""
+        return n_atoms(self.trajectory)
 
-    @property
-    def D_J(self) -> 'uravu.distribution.Distribution':
-        """
-        :return: Jump diffusion coefficient
-        """
-        return self._diff.D_J
 
-    @property
-    def flatchain(self) -> np.ndarray:
-        """
-        :return: sampling flatchain
+    def jump_diffusion(self, start_dt: sc.Variable, diffusion_params: Union[dict, None] = None) -> None:
         """
-        return np.array([self.D_J.samples, self.intercept.samples]).T
+        Calculate the jump diffusion coefficient using the mean-squared total displacement data.
+        
+        :param start_dt: The time at which the diffusion regime begins.
+        :param diffusion_params: The keyword arguements for the diffusion calculation
+            (see :py:func:`diffusion.bayesian_regression`). Optional, defaults to :py:attr:`None`.
+        """
+        if diffusion_params is None:
+            diffusion_params = {}
+        self.diff = Diffusion(msd = self.mstd, n_atoms = self.n_atoms)
```

Comment:
```suggestion
        self.diff = Diffusion(msd=self.mstd, n_atoms=self.n_atoms)
```

Might be easier to run things through the linter. Typically we use 

```
yapf -i kinisi/*.py
``` 

which does the linting inplace (you might need to install `yapf`).